task/gcp/resources/data_source_bucket.go

```diff
@@ -0,0 +1,51 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+
+	"terraform-provider-iterative/task/common"
+	"terraform-provider-iterative/task/common/machine"
+)
+
+// NewExistingBucket creates a new data source referring to a pre-allocated GCP storage bucket.
+func NewExistingBucket(clientCredentials string, storageParams common.RemoteStorage) *ExistingBucket {
```

Comment:
(Off-topic) `NewExisting` ...I get the meaning, of course, but still tickles my cache invalidation sense.

Reply:
It's pretty much a convention thing at this point. Happy to rename if you like.

Reply:
I may eventually propose explicitly differentiating resources and data sources, but that's for a separate pull request.